### PR TITLE
bt-232: Connect signIn to api

### LINKED ISSUE #232: Connect SignIn to API
User story:

As a Talent, I want to Sign In to BSA hiring platform, so that can use all the functionalities as a registered user for looking for a job

Acceptance criteria:

- If the Talent enters the Email and Password used for registration and clicks Login is redirected to the page “Chat”
- If the fields Email and Password are empty, these fields light red color, and tips with information that this field is required appear, the user stays on this page
- After entering invalid data (does not match validation requirements) the notification about an invalid Email or Password appears on the screen, these fields light red color, and the user stays on this page
- If entered Email and Password are not before used the notification “User does not exist” appears, Talent stays on this page
- If the data in the field Password are invalid the notification about the possibility of password recovery appears with the link to the page  to recover the password
- Click on the button “Forgot password” opens the page to recover the password 
- Click on the link “Create an account” redirects to the page “Sign Up”

User story:

As an Employer, I want to Sign In to BSA hiring platform, so that can use all the functionalities as a registered user for hiring employees

Acceptance criteria:

- After entering Email and Password used for registration and  clicking Login the Employer is redirected to the page “Search”
- Other points are the same as Talent’s role

### PR DESCRIPTION
**Pull request by vladjsp**

### CODE REVIEW

**Comment by vladjsp** [comment]
signIn

**Comment by vladjsp** [comment]
https://github.com/BinaryStudioAcademy/bsa-2023-bsa-talents/assets/118971905/4106aed1-3dd5-4ac0-a0c8-29d176ae36b3

**Review by JosuerBague — APPROVED** [review]
Tested locally on my machine.

Everything looks correct and working.

File: frontend/src/bundles/auth/store/slice.ts [hunk 1]
```diff
@@ -21,25 +21,26 @@ const { reducer, actions, name } = createSlice({
...
+                (state) => {
+                    state.dataStatus = DataStatus.PENDING;
+                },
+            )
+            .addMatcher(
+                isAnyOf(signUp.rejected, loadUser.rejected, signIn.rejected),
+                (state) => {
+                    state.dataStatus = DataStatus.REJECTED;
+                },
+            )
```
Comment (mskmee) [inline frontend/src/bundles/auth/store/slice.ts:37]: Guess we should set current user to null in this matchers

File: frontend/src/bundles/auth/store/slice.ts [hunk 2]
```diff
@@ -21,25 +21,26 @@ const { reducer, actions, name } = createSlice({
...
+            .addMatcher(
+                isAnyOf(signUp.fulfilled, loadUser.fulfilled, signIn.fulfilled),
+                (state, action) => {
+                    state.dataStatus = DataStatus.FULFILLED;
+                    state.currentUser = action.payload;
```
Comment (nikita-remeslov) [inline frontend/src/bundles/auth/store/slice.ts:42]:
```suggestion
                    state.currentUser = null;
```

**Review by nikita-remeslov — CHANGES_REQUESTED** [review]

File: frontend/src/bundles/auth/store/slice.ts [hunk 3]
```diff
@@ -21,25 +21,26 @@ const { reducer, actions, name } = createSlice({
...
+            .addMatcher(
+                isAnyOf(signUp.pending, loadUser.pending, signIn.pending),
+                (state) => {
+                    state.dataStatus = DataStatus.PENDING;
+                },
+            )
+            .addMatcher(
+                isAnyOf(signUp.rejected, loadUser.rejected, signIn.rejected),
+                (state) => {
+                    state.dataStatus = DataStatus.REJECTED;
```
Comment (nikita-remeslov) [inline frontend/src/bundles/auth/store/slice.ts:34]:
```suggestion
                    state.dataStatus = DataStatus.REJECTED;
                    state.currentUser = null;
```

**Review by nikita-remeslov — APPROVED** [review]

**Pull request merged**